docs/source/rdump.rst

```diff
@@ -0,0 +1,115 @@
+rdump
+-----
+
+By default, target-query generates records. Records in this sense are binary representations of parsed artefacts.
+They are transformed to text by the default mechanism. An example is given below:
+
+.. code-block:: console
+
+    <record key1="value1" key2="value2" >
+
+With ``rdump`` you can transform the stream of records to your liking. The ``rdump`` utility allows you to:
+
+* Select certain fields from the records.
+* Filter certain records.
+* Create additional derived fields.
+* Limit the output.
+* Format the results.
+* Write the results through an adapter.
+
+.. hint::
+
+    Don't know yet what a record is? Read more :ref:`here <overview/index:records>` for a detailed explanation.
+
+Field Selection
+~~~~~~~~~~~~~~~
+
+Imagine we're going to extract a list of users from a forensic image using:
+
+.. code-block:: console
+
+    $ target-query host.img -f users 
+
+To select only the name and home fields we use the ``-F`` option:
+
+.. code-block:: console
+
+    $ target-query host.img -f users | rdump -F name,home
+
+Using the ``-X`` option we can exclude fields:
+
+    $ target-query host.img -f users | rdump -X hostname,domain,sid
+
+
+Filtering
+~~~~~~~~~
+
+Filtering records can be done through the ``-s`` option.
+The selection option must be a Python-expression, where the record is represented with the symbol ``r``.
+So to eliminate all records that for example have no domain value:
+
+.. code-block:: console
+
+    $ target-query host.img -f users | rdump -F name,home -s "r.domain is not None"
+
+
+Derivations
+~~~~~~~~~~~
+
+It is possible to create new fields derived from other fields. This can be done by using
+an expression and the ``-E`` option. The following example lists all dll files from
+a host and adds the filename to a separate field:
+
```

Comment:
do you have to make sure you don't use a name of an existing field? how do you know which fields are included in the record?

Reply:
It is quite obvious that you can use target-query for this. Also, yes, all fields must exist. This is common knowledge.